src/main/java/net/dv8tion/jda/internal/requests/restaction/RoleActionImpl.java

```diff
@@ -167,8 +203,17 @@ protected RequestBody finalizeData()
         DataObject object = DataObject.empty();
         if (name != null)
             object.put("name", name);
-        if (color != null)
-            object.put("color", color & 0xFFFFFF);
+        if (colors != null)
+        {
+            object.put("color", colors.getPrimaryColorRaw() == Role.DEFAULT_COLOR_RAW ? 0 : colors.getPrimaryColorRaw() & 0xFFFFFF);
+
+            DataObject colorsObject = DataObject.empty();
+
+            colorsObject.put("primary_color", colors.getPrimaryColorRaw() == Role.DEFAULT_COLOR_RAW ? 0 : colors.getPrimaryColorRaw() & 0xFFFFFF);
+            colorsObject.put("secondary_color", colors.getSecondaryColorRaw() == null ? null : colors.getSecondaryColorRaw() & 0xFFFFFF);
+            colorsObject.put("tertiary_color", colors.getTertiaryColorRaw() == null ? null : colors.getTertiaryColorRaw() & 0xFFFFFF);
+            object.put("colors", colorsObject);
```

Comment:
Dont Repeat Yourself, move this to `RoleColorsImpl#toData()` (make `RoleColorsImpl` implement `SerializableData`)